tests/run/legacy_implicit_noexcept.pyx

```diff
@@ -46,55 +46,51 @@ cdef test_noexcept_warning():
 def func_pure_implicit() -> cython.int:
     raise RuntimeError
 
-@cython.excetval(check=False)
+@cython.exceptval(check=False)
 @cython.cfunc
 def func_pure_noexcept() -> cython.int:
     raise RuntimeError
 
-def return_stderr(func):
+def print_stderr(func):
     @functools.wraps(func)
     def testfunc():
-        old_stderr = sys.stderr
-        stderr = sys.stderr = StringIO()
-        try:
+        from contextlib import redirect_stderr
```

Comment:
Contextlib doesn't seem to be working on Python 2.7 on the 3.0.x cherry-pick unfortunately.

Reply:
I committed directly to the branch for simplicity and here is result :-(.

Reply:
I have created a PR to 3.0.X reverting the changes: #6103. But I don't have python2.7 anymore so cannot test them locally. I will have more time playing with it in the eventing (I hope so).

Reply:
No worries. I don't think a temporarily failing branch will cause real problems.